compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/window/Popup.skiko.kt

```diff
@@ -392,11 +390,15 @@ fun Popup(
     onKeyEvent: ((KeyEvent) -> Boolean)? = null,
     content: @Composable () -> Unit
 ) {
+    val currentOnDismissRequest by rememberUpdatedState(onDismissRequest)
```

Comment:
As far as I remember, we don't have to use `rememberUpdatedState` if a parameter used in lambda passed to Composable. Because it is recreated every time in a new composition.

Did you test how it behaves?

Reply:
Not sure I understand. The point of using `rememberUpdatedState` here is exactly to avoid recreating the lambda on every recomposition.

Reply:
From my experience, it was used mainly to pass the updated state of a new parameter in a long-lived lambda (that is created in first composition). I.e. in:
```
val obj = remember {
  SomeOjb(onDismiss = { currentOnDismissRequest() })
}
```
```
Modifier.pointerInput(Unit) {
  ...
  currentOnDismissRequest()
  ...
}
```

The your case is new to me.

Compose Compiler won't recreate `overriddenOnKeyEvent`?

Reply:
> From my experience, it was used mainly to pass the updated state of a new parameter in a long-lived lambda (that is created in first composition)

That's basically the same thing as here: `overriddenOnKeyEvent` is the long-lived lambda.

> Compose Compiler won't recreate overriddenOnKeyEvent?
```
var recompositionCount = 0
var clickedValue = 0

@Composable
fun MyButton(onClick: () -> Unit) {
    recompositionCount += 1
    Button(onClick = onClick, Modifier.testTag("button")) {
        Text("Button")
    }
}

@Composable
fun Content(n: Int) {
    val currentN by rememberUpdatedState(n)
    MyButton(
        onClick = { clickedValue = currentN },
    )
}

class UpdatedStateTest {

    @OptIn(ExperimentalTestApi::class)
    @Test
    fun testUpdatedStateRecomposition() = runComposeUiTest {
        var n by mutableStateOf(0)

        setContent {
            Content(n)
        }

        assertEquals(1, recompositionCount)
        n = 10
        onNodeWithTag("button").performClick()
        assertEquals(1, recompositionCount)
        assertEquals(10, clickedValue)
    }

}
```

Reply:
Interesting, thanks. I will think about it more. I was under impression that Compose Compiler should either optimise it, or it isn't and performance issue.

Anyway, this comment wasn't about issue issue, just about potential redundancy.